### PR TITLE
PeoplePicker: Adding custom validator and styling adjustment

### PR DESCRIPTION
**Pull request by kristoferbrown**
#### Pull request checklist
- [x] Include a change request file using `$ npm run change`

#### Description of changes

**Callout**
 - Removing border to make callout match Fabric toolkit and prevent undesired side-effects in the people picker suggestion callout.

**Persona**
- Styling adjustments to further align with the Fabric toolkit. 
- Decreasing height of primary text to allow for error state on persona pills.

**PeoplePicker**
- Styling adjustments to further align with the Fabric toolkit.
- Made delete keypresses detect if they should delete a suggestion or a selected person.
- Added styling for invalid persona pills.
- Added onValidateInput prop and validationState enum.
- Allowed arbitrary text to create new selected persona pills, but only if they meet the requirements set by whatever function was passed to onValidateInput.

**PeoplePicker Example**
- Added a basic example validator function, it only checks if you've typed more than a single character or if you included and @ symbol.
- Removed 0 key, it was causing issues with 0 evaluating falsey when deciding to use provided keys or indexes as keys.

#### Focus areas to test

Persona and People Picker.

Testable at the following devlink:
[https://fabricreact.azurewebsites.net/fabric-react/v-krbrow-pc-v-krbrow-peoplepicker-bugfixes/#/examples/PeoplePicker](https://fabricreact.azurewebsites.net/fabric-react/v-krbrow-pc-v-krbrow-peoplepicker-bugfixes/#/examples/PeoplePicker
)

picker-validator

### CODE REVIEW

**Comment by dzearing** [comment]
maybe some screenshots of what you're doing?

**Comment by kristoferbrown** [comment]
@dzearing, I just added a dev link and a screenshot.

**Comment by dzearing** [comment]
@kristoferbrown can you get William and Kathy's github accounts and make sure they see the PR?

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts [hunk 1]
```diff
@@ -125,4 +129,10 @@ export interface IBasePickerSuggestionsProps {
    * Indicates whether to show a button with each suggestion to remove that suggestion.
    */
   showRemoveButtons?: boolean;
+}
+
+export enum validationState {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts]: Nit: I believe all of our enums start with capitol letters so this should be ValidationState.

File: packages/office-ui-fabric-react/src/components/pickers/Suggestions/SuggestionsController.ts [hunk 2]
```diff
@@ -81,6 +87,22 @@ export class SuggestionsController<T> {
     this.suggestions.splice(index, 1);
   }
 
+  public createGenericSuggestion(name: string, currentValidationState: validationState) {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/Suggestions/SuggestionsController.ts]: SuggestionController isn't just for peoplepickers, it's for all pickers. You shouldn't call getInitials here. CreateGenericSuggestion should probably be another pass through renderer like RenderItem

**Comment by kristoferbrown** [comment]
@joschect, take a look at the latest commit. It keeps some of createGenericSuggestion in SuggestionController but makes it use a customizable prop instead.

**Pull request merged**